### DISCUSSION TITLE
Project management

### DISCUSSION THREAD

**Discussion by LucaTheHacker** [immich-app/immich#2941] [General]
I noticed that Immich currently uses just one repo as a monolithic repo.
I think that splitting the project in different repositories would make easier for third-parties to help developing Immich.

Is there any specific reason why the backend is in the same repo as the mobile application or it is just "always been like that"?

**Comment by etnoy** [comment]
Monorepos have many advantages over split repos. What specific advantage do you see with splitting?

**Reply by LucaTheHacker** [reply]
Finding things is way easier when things are divided. 
If I need to work to the backend, I don't need to have files from app hanging around my IDE, it just makes harder for the IDE to handle all the files at once and makes harder for the human to only check the right place.

Surely you can open with your IDE a defined subdirectory, but then you'll lose most of the git functionality that is indeed good.

**Reply by alextran1502** [reply]
We are using openapi to generate sdk for the clients (web and mobile). And in many occasions your work is a combination of web/server or mobile/server. Having these project in the same repo helps with one command (`make dev`) to start up everything for nice developer experience

**Comment by uhthomas** [comment]
This is a never ending conversation and there is no right answer. I think it depends a lot on context, and for Immich I think a single repository makes the most sense. Immich is a small team of volunteers and the overhead of managing multiple repositories would be a huge negative impact on productivity. Less active components will also be more likely to rot, as we have already observed with [the cli](https://github.com/immich-app/CLI). If we can barely manage two repositories, I struggle to see how we could manage 8 or more repositories.

I currently do not see much pain from using a monorepo and really only see positives. It also makes cross-component changes (i.e web and server) really easy.

The amount of effort required to split Immich into multiple repositories just doesn't make sense to me, and I think it would kill velocity with increased complexity. We're actively looking to merge more things into this repository (again, the cli) because it will be much easier to manage and distribute.
